b2/sync/sync.py

```diff
@@ -210,6 +218,9 @@ def sync_folders(
     if dest_folder.folder_type() == 'local' and not dry_run:
         dest_folder.ensure_present()
 
+    if source_folder.folder_type() == 'local' and not dry_run and fail_on_empty_source:
```

Comment:
Any reason to skip the check on a dry run?

Reply:
Nope, I was just following the convention of the target directory check above. I'll remove the dry_run check.